p34/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+use std::ops;
+
+#[derive(Debug)]
+struct BigUint4096 {
+    values: [u64; 64],
+}
+
+impl PartialEq for BigUint4096 {
+    fn eq(&self, other: &Self) -> bool {
+        for i in 0..64 {
+            if self.values[i] != other.values[i] {
+                return false;
+            }
+        }
+        true
+    }
+}
+
+impl Eq for BigUint4096 {}
+
+impl ops::Add<BigUint4096> for BigUint4096 {
```

Comment:
It's worth to add implementations for references as well.

Reply:
solved